### PR TITLE
Fix/13573 wiki toolbar entry

### PR DESCRIPTION
**Pull request by tessi**
This is a WIP PR just now. I still have to watch (and add) tests.

Fixes: https://www.openproject.org/work_packages/13573

When saving a work package query under a new name (and the old query was starred), the new querry was starred but didn't get a menu item.

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage. <a href='http://fix-13573-wiki-toolbar-entry.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Click to open.</a>

File: spec/features/work_packages/saving_queries.rb [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  let(:role) { FactoryGirl.create :role, :permissions => [:view_work_packages] }
+  let(:current_user) { FactoryGirl.create :user, member_in_project: project,
+                                                 member_through_role: role,
+                                                 admin: true }
+
+  let(:filter_name) { 'done_ratio' }
+  let!(:query) do
+    query = FactoryGirl.build(:query, project: project, is_public: true)
+    query.filters = [Queries::WorkPackages::Filter.new(filter_name, operator: ">=", values: [10]) ]
+    query.save and return query
```
Comment (myabc) [inline spec/features/work_packages/saving_queries.rb]: (I don't agree with Hound here)

Comment (tessi) [inline spec/features/work_packages/saving_queries.rb]: I changed it to

``` ruby
query.save!
return query
```

so that everyone can feel good again

Comment (myabc) [inline spec/features/work_packages/saving_queries.rb]: no need for an explicit `return` though

Comment (tessi) [inline spec/features/work_packages/saving_queries.rb]: fair enough

**Pull request merged**